lib/chef/provider/package/apt.rb

```diff
@@ -89,6 +89,7 @@ def locked_packages
         end
 
         def install_package(name, version)
+          name = check_availability(name)
```

Comment:
This only fixes... part of the bug.

It will ensure that if you try to purge `['a', 'b', 'c']` and `c` isn't a known package, that it won't try to purge `c`... which is good...

But the actual bug is that if `a` is installed, it'll automatically try to uninstall `b`, and `c` even if they are not installed.

So imagine:
* a - installed
* b - uninstalled, but available
* c - uninstalled and unavailable

Current behavior is Chef will try to purge `a`, `b` and `c`

With your PR it'll just try to purge `a` and `b`

But the *right* behavior is that it should only try to purge `a`.